vars/buildMessage.groovy

```diff
@@ -6,11 +6,20 @@
  * this file be licensed under the Apache-2.0 license or a
  * compatible open source license.
  */
+ /** Library to find a pattern in jenkins build log.
+ @param Map args = [:] args A map of the following parameters
+ @param args.search <required> - Use 'pass' to get the components passed and 'fail' for components failed.
+ */
 import com.cloudbees.groovy.cps.NonCPS
 import org.apache.commons.io.IOUtils
 @NonCPS
-def call(){
-    String ERROR_STRING = "Error building"
+def call(Map args = [:]){
+    String QUERY_STRING = ''
+    if (args.search == 'fail') {
+       QUERY_STRING = "Error building"
+    } else if (args.search == 'pass') {
+       QUERY_STRING = "Build successful"
+    }
```

Comment:
Who not take entire search query as arg? Reason being if we change the query in jenkins CI which is coming from here https://github.com/opensearch-project/opensearch-build/blob/e5e214da1b5b423977e211dd8c0a6d3205451058/src/run_build.py#L71 we might miss it.

Reply:
We are already taking the entire log sayali and filtering for `Error building`, this denotes that the component is failed. Yes we need to come back and change if the log statement changed.

Reply:
Can we add a check if the args is either `pass` or `fail` so that any other string does not go through?

Reply:
I see your point, then we can make it generic and while using in the jenkinsfile we can use `buildMessage(search: 'Error building')` or `buildMessage(search: 'Build successful')`, is that what ur referring to @gaiksaya ?

Reply:
Yes Exactly!

Reply:
Updated in my latest commit @gaiksaya.